python/activator/middleware_interface.py

```diff
@@ -1595,7 +1653,7 @@ def _get_safe_dataset_types(butler):
 
     @connect.retry(2, DATASTORE_EXCEPTIONS, wait=repo_retry)
     def _export_subset(self, exposure_ids: set[int],
-                       dataset_types: typing.Any, in_collections: typing.Any) -> None:
+                       dataset_types: typing.Any, in_collections: typing.Any) -> list[DatasetRef]:
```

Comment:
Note that the docs spec this as a generic collection. As stated above, I think that's the right level of abstraction, but the method should at least be self-consistent.